### ISSUE TITLE
Memory Leak?

### ISSUE THREAD

**Issue by boazsegev** [ohler55/agoo#9]
I got up to a couple of Giga bytes when running a simple Hello World application...

This is my Rack application (`config.ru`):

```ruby
module HelloWorld
  # This is the HTTP response object according to the Rack specification.
  RESPONSE = [200, { 'Content-Type' => 'text/html',
          'Content-Length' => '12' },
   ['Hello World!']]

   # this is function will be called by the Rack server (iodine) for every request.
   def self.call env
     # simply return the RESPONSE object, no matter what request was received.
     RESPONSE
   end
end

# this function call links our HelloWorld application with Rack
run HelloWorld
```

I ran Agoo using rackup:

```bash
$ RACK_ENV=production rackup -p 3000 -r agoo -s Agoo
```

I benchmarked the server with `wrk`:

```bash
wrk -c40 -d15 -t2 http://localhost:3000/
```

After a single benchmark on macOS 10.12.6 , I got **memory consumption up to 3.33Gig**.

I think the server is leaking memory or maintaining pointers to obsolete Ruby objects (the response isn't creating new objects, but maybe the handles cost memory).

**Comment by ohler55** [comment]
I look at that. My earlier tests were stable but sounds like something is off.

**Comment by boazsegev** [comment]
I just tested the 1.2.2 release and everything works great.

Nice work one the quick fix!

**Comment by ohler55** [comment]
Yeah, was not freeing the requests. Thanks for catching it. Now I can get back to our pub-sub discussion.

**Comment by ohler55** [comment]
I believe this is fixed in 2.0.0.
